android-pdf-viewer/src/main/java/com/github/barteksc/pdfviewer/annotation/core/pdf/PdfUtil.kt

```diff
@@ -0,0 +1,367 @@
+                            for (j in 0 until relationsArray.size()) {
+                                val documentationDict: PdfDictionary = relationsArray.getAsDict(j)
+                                val schemaId: PdfNumber? =
+                                    documentationDict.getAsNumber(PdfName("schemaId"))
+                                val documentId: PdfString? =
+                                    documentationDict.getAsString(PdfName("documentId"))
+
+                                if (schemaId != null && documentId != null) {
+                                    logDebug(TAG, "Annotation $i - Relations $j:")
+                                    logDebug(TAG, "  schemaId: $schemaId")
+                                    logDebug(TAG, "  documentId: $documentId")
+
+                                    documentations.add(
+                                        Documentation(
+                                            schemaId.intValue().toLong(),
+                                            documentId.toString()
+                                        )
+                                    )
+                                }
+
+                            }
+                        }
+                        val squareAnnotation = Annotation(
+                           AnnotationType.SQUARE.name,
+                            squareAnnotationPoints,
+                            relations = Relations(documentations)
+                        )
+                        annotationsList.add(squareAnnotation)
+
+                        logDebug(TAG, "Annotation is square")
+                        logDebug(TAG, "Annotation $i on page $pageNum - points:")
+                        logDebug(TAG, "squareAnnotationPoints:$squareAnnotationPoints")
+
+                    } else if (subtype == PdfName.CIRCLE) {
+                        // bottom left
+                        val xBottomLeftPoint = llx
+                        val yBottomLeftPoint = lly
+                        val bottomLeftPoint = PointF(xBottomLeftPoint, yBottomLeftPoint)
+
+                        // top right
+                        val xTopRightPoint = urx
+                        val yTopRightPoint = ury
+                        val topRightPoint = PointF(xTopRightPoint, yTopRightPoint)
+
+                        val circleAnnotationPoints = listOf(bottomLeftPoint, topRightPoint)
+                        val circleAnnotation = Annotation("CIRCLE", circleAnnotationPoints)
+                        annotationsList.add(circleAnnotation)
+
+                        logDebug(TAG, "Annotation is circle")
+                        logDebug(TAG, "Annotation $i on page $pageNum - points:")
+                        logDebug(TAG, "circleAnnotationPoints:$circleAnnotationPoints")
+                    } else {
+                        logDebug(TAG, "Annotation is not recognised")
+                    }
+                }
+            }
+        }
```

Comment:
This is better of to be a 

```kotlin
when(subtype) {
  Rectangle -> {
/// handle rectangle
}
else -> emptyList()
}
```
make sure to extract rectangle handling in a seperate method to improve readability. Also extraction of relations should be a seperate method as well cause thats gonna be reused across all the shapes I find

Reply:
Extracted